app/Http/Controllers/SearchController.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Resource;
+use Illuminate\Http\Request;
+
+class SearchController extends Controller
+{
+	/**
+	 * Whether we should return JSON instead of a view.
+	 * @var bool
+	 */
+	protected $return_JSON;
+
+	/**
+	 * Whether to interpret all queries literally by escaping reserved characters.
+	 * Setting this value to true will disable fancy searching.
+	 * @var bool
+	 */
+	protected $literal_queries = false;
+
+	/**
+	 * At most how many hits should we return?
+	 * @var integer
+	 */
+	protected $number_of_hits = 20;
+
+	/**
+	 * A list of all available fields and their contributions to the overall search score.
+	 * A field's weight will default to 0 if not specified, so specify all of them!
+	 * Set this variable to a falsey value to assign all fields the same weight of 1.
+	 * @var array
+	 */
+	protected $field_weights = [
```

Comment:
We should check that these are set to the appropriate values. My main concern is that the fractional weights will make it so that no results will appear, especially since search results need to meet a score threshold of at least 2 to show up in the results.

Reply:
I tested the field weights in c1ff2dd. They seem to work fine, even with the high score threshold. Including a content type (like "text" or "link") will not significantly skew the ordering of your average search query, but performing a query with only that one word returned only resources with that content type. I'll call that a success.